src/batchUploader.ts

```diff
@@ -129,21 +131,68 @@ export class BatchUploader {
         return offlineStoragePercentage >= rampNumber;
     }
 
+    private shouldDebounceAST(): boolean {
+        const now = Date.now();
+        if (now - this.lastASTEventTime < this.AST_DEBOUNCE_MS) {
+            return true;
+        }
+
+        this.lastASTEventTime = now;
+
+        return false;
+    }
+
+    private createBackgroundASTEvent(): SDKEvent {
```

Comment:
I will DRY this and the code in [servermodels](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/serverModel.ts#L340-L390) up, but wanted to get a POC going first.

Reply:
Make a sustained eng ticket please.